### PR TITLE
Dqueue

### PR DESCRIPTION
**Pull request by sahibak**
Code for dqueue

### CODE REVIEW

File: library/dqueue.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    if (this.count === 0) {
+      throw new Error("Error");
+    }
+    this.tail += 1;
+    this.tail = this.tail % this.lengthOfDqueue;
+    let poppedValue = this.dqueue[this.tail];
+    this.dqueue[this.tail] = null;
+    this.count -= 1;
+    this.resetValuesOnEmpty();
```
Comment (yagnik) [inline library/dqueue.js:104]: why was this needed, won't it continue to work no matter where the pointers are ?

Comment (sahibak) [reply library/dqueue.js]: Not needed, as the mod function will perform the resetting. however resetting is required, otherwise pointers will go out of range of length.
